cmd/compute-domain-controller/computedomain.go

```diff
@@ -215,9 +219,38 @@ func (m *ComputeDomainManager) AssertWorkloadsCompleted(ctx context.Context, cdU
 	}
 
 	if len(nodes.Items) != 0 {
+		// show nodes with labels
+		nodeNames := []string{}
+		for _, node := range nodes.Items {
+			nodeNames = append(nodeNames, node.Name)
+		}
+		klog.Errorf("Found %d nodes with label for ComputeDomain with UID  %s: %v", len(nodes.Items), cdUID, nodeNames)
 		return fmt.Errorf("nodes exist with label for ComputeDomain %s", cdUID)
 	}
 
+	// check if all resource claims for workloads are gone
+	cd, err := m.Get(cdUID)
+	if err != nil {
+		return fmt.Errorf("error retrieving ComputeDomain: %w", err)
+	}
+
+	resourceClaims, err := m.config.clientsets.Core.ResourceV1beta1().ResourceClaims(cd.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: metav1.FormatLabelSelector(labelSelector),
+	})
+	if err != nil {
+		return fmt.Errorf("error retrieving ResourceClaims: %w", err)
+	}
+
+	if len(resourceClaims.Items) != 0 {
+		claimNames := []string{}
+		for _, claim := range resourceClaims.Items {
+			claimNames = append(claimNames, claim.Name)
+		}
+		klog.Errorf("Found %d ResourceClaims for ComputeDomain with UID %s: %v",
+			len(resourceClaims.Items), cdUID, claimNames)
+		return fmt.Errorf("ResourceClaims exist for ComputeDomain %s", cdUID)
+	}
+
```

Comment:
This is not related to adding logs and should be in its own PR.

That said, as currently implemented this is a NOOP -- the ResourceClaims generated from our ResourceClaimTemplate for workloads will never have the `ComputeDomain` label applied to them, so the list operation will *always* return an empty list.

Reply:
Yes agree. this should be in a separate PR. I just added this here after i added the iterations on node.Names. It was based on the list of nodes logic. And anyway i spent more time adding this part than the logs itself. 

Oh, i checked this https://github.com/NVIDIA/k8s-dra-driver-gpu/blob/main/templates/compute-domain-daemon-claim-template.tmpl.yaml#L10 and thought this label will be able in all the resourceclaim templates 

Oh i see, its an empty field at the time of generation.

Reply:
This comment was actually more relevant on an older iteration of the code. The right way to guarantee that all workloads have stopped now is to wait until no *nodes* have the ComputeDomain label anymore. This is sufficient because the workload pods remove this label as they shutdown.